conans/client/graph/graph_binaries.py

```diff
@@ -393,21 +395,26 @@ def _evaluate_single(n):
 
         levels = deps_graph.by_levels()
         config_version = self._config_version()
+        # TODO: Find a better way to define this number, we ended up with 8 because
+        # it seemed a goof fit from local testing - But this could be improved
+        # max node count in every level/user conf?
+        evaluate_pool = ThreadPool(8)
         for level in levels[:-1]:  # all levels but the last one, which is the single consumer
             for node in level:
                 self._evaluate_package_id(node, config_version)
-            # group by pref to paralelize, so evaluation is done only 1 per pref
+            # group by pref to parallelize, so evaluation is done only 1 per pref
             nodes = {}
             for node in level:
                 nodes.setdefault(node.pref, []).append(node)
-            # PARALLEL, this is the slow part that can query servers for packages, and compatibility
-            for pref, pref_nodes in nodes.items():
-                _evaluate_single(pref_nodes[0])
-            # END OF PARALLEL
+            # Evaluate the first node of each pref
+            evaluate_pool.map(_evaluate_single, [pref_nodes[0] for pref_nodes in nodes.values()])
+
             # Evaluate the possible nodes with repeated "prefs" that haven't been evaluated
             for pref, pref_nodes in nodes.items():
                 for n in pref_nodes[1:]:
                     _evaluate_single(n)
+        evaluate_pool.close()
+        evaluate_pool.join()
```

Comment:
Shouldn't the ``join`` be done after the ``.map()``? Otherwise the evaluation of the lines 413-415 will be wrong?

Reply:
As per https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.map

> It blocks until the result is ready.

The `join` here is unecessary as far as I can tell, but I left it out of habit

Reply:
Oh, yes, you are right, the ``map()`` is blocking, I got confused by the ``join()``, which is unnecessary and I think it can be removed.